terraso_backend/apps/export/graphql/mutations.py

```diff
@@ -0,0 +1,155 @@
+# Copyright © 2021-2025 Technology Matters
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see https://www.gnu.org/licenses/.
+
+import graphene
+from django.contrib.auth import get_user_model
+from graphql import GraphQLError
+
+from apps.project_management.models import Project, Site
+
+from ..models import ExportToken
+from .types import ExportToken as ExportTokenType
+from .types import ResourceTypeEnum
+
+User = get_user_model()
+
+
+def get_user_tokens(user):
+    """Get all export tokens for the current user."""
+    return ExportToken.objects.filter(user_id=str(user.id))
+
+
+def can_create_export_token(user, resource_type, resource_id):
+    """
+    Check if user has permission to CREATE an export token for a resource.
+    User must have access to the underlying resource.
+
+    Rules:
+    - USER: Only the user themselves
+    - PROJECT: Any project member
+    - SITE: Site owner (unaffiliated) or any project member (project sites)
+    """
+    if user.is_anonymous:
+        return False
+
+    if resource_type == "USER":
+        # Users can only create tokens for themselves
+        return str(user.id) == resource_id
+
+    elif resource_type == "PROJECT":
+        # Any project member can create their own token
+        try:
+            project = Project.objects.get(pk=resource_id)
+            return project.membership_list.memberships.filter(
+                user=user,
+                deleted_at__isnull=True,
```

Comment:
question: did you try without the `deleted_at__isnull` check and it didn't work? i thought the soft delete functionality handles this stuff automatically, but i know there are other places in the app where we use a check like this, and i've never understood when/why it is necessary.